File: 1642. Water Bottles/Solution.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        r = numBottles
        while(numBottles>=numExchange):
            temp = numBottles//numExchange
            tempr = numBottles%numExchange
            r +=temp
            numBottles = temp+tempr
            # print(numBottles,temp,tempr,r)
        
        return int(r)

File: 1642. Water Bottles/Solution.cpp
class Solution {
public:
    int numWaterBottles(int b, int e) {
        int count=b;
        int temp=0;
        int carry=0;
        while(b>=e)
        {
            carry=b%e;
            temp=b/e;
            count+=temp;
            b=temp+carry;
        }
        return count;
    }
};